### ISSUE TITLE
Remove register assignment direction queries

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#2450]
Each code generator provides the ability to define and query the direction the register assigner is proceeding (either forwards or backwards).  All of the code generators do backwards register assignment (and have for nearly 20 years) and don't use these queries.  Remove the functions:
```
getAssignmentDirection()
setAssignmentDirection()

and the RegisterAssignmentDirection enum
```
from the following code generators:
- [ ] Power
- [ ] Z
- [ ] ARM

x86 is the exception as it assigns X87 registers in a forward pass.  Although general X87 support is deprecated, I can't say for certain that something won't break if the forward pass is removed at this time.  However, it can be removed once the X87 code is removed (#946).
- [ ] X86

**Comment by vignesh240** [comment]
I would like to fix this issue @0xdaryl

**Comment by vignesh240** [comment]
Thank you for baring with me, I have done with the corrections let me know if I have to do any changes

**Comment by fjeremic** [comment]
This was fixed in #2493, closing the issue.
